src/transformers/models/falcon_vlm/modeling_falcon_vlm.py

```diff
@@ -0,0 +1,898 @@
+            don't have their past key value states given to this model) of shape `(batch_size, 1)` instead of all
+            `decoder_input_ids` of shape `(batch_size, sequence_length)`.
+        inputs_embeds (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`, *optional*):
+            Optionally, instead of passing `input_ids` you can choose to directly pass an embedded representation. This
+            is useful if you want more control over how to convert `input_ids` indices into associated vectors than the
+            model's internal embedding lookup matrix.
+        vision_feature_layer (`int`, *optional*, defaults to -2):
+            The index of the layer to select the vision feature.
+        vision_feature_select_strategy (`str`, *optional*, defaults to `"default"`):
+            The feature selection strategy used to select the vision feature from the vision backbone.
+            Can be one of `"default"` or `"full"`. If `"default"`, the CLS token is removed from the vision features.
+            If `"full"`, the full vision features are used.
+        use_cache (`bool`, *optional*):
+            If set to `True`, `past_key_values` key value states are returned and can be used to speed up decoding (see
+            `past_key_values`).
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+@add_start_docstrings(
+    """The FalconVlm model which consists of a vision backbone and a language model.""",
+    FALCON_VLM_START_DOCSTRING,
+)
+# Copied from transformers.models.llava_next.modeling_llava_next.LlavaNextForConditionalGeneration with LlavaNext->FalconVlm,llava_next->falcon_vlm
+class FalconVlmForConditionalGeneration(FalconVlmPreTrainedModel):
+    # Ignore copy
+    def __init__(self, config: FalconVlmConfig):
+        super().__init__(config)
+
+        self.vision_tower = AutoModel.from_config(config.vision_config)
+
+        self.mm_projector = FalconVlmMultiModalProjector(config)
+
+        self.image_newline = nn.Parameter(torch.randn(config.text_config.hidden_size, dtype=self.dtype))
+
+        self.vocab_size = config.text_config.vocab_size
+
+        self.language_model = AutoModelForCausalLM.from_config(
+            config.text_config, attn_implementation=config._attn_implementation
+        )
+
+        self.pad_token_id = self.config.pad_token_id if self.config.pad_token_id is not None else -1
+        self.post_init()
+
+    @property
```

Comment:
? not sure I understand what this is

Reply:
By that I just mean that it can just be an attribute, let' s not overcomplicate. 
And I would even just remove it, it's not super relevant